src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryToActivityConverter.cs

```diff
@@ -0,0 +1,120 @@
+﻿// <copyright file="RedisProfilerEntryToActivityConverter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using OpenTelemetry.Trace;
+using StackExchange.Redis.Profiling;
+
+namespace OpenTelemetry.Instrumentation.StackExchangeRedis.Implementation
+{
+    internal static class RedisProfilerEntryToActivityConverter
+    {
+        public static Activity ProfilerCommandToActivity(Activity parentActivity, IProfiledCommand command)
+        {
+            var name = command.Command; // Example: SET;
+            if (string.IsNullOrEmpty(name))
+            {
+                name = "name";
+            }
+
+            var activity = StackExchangeRedisCallsInstrumentation.ActivitySource.StartActivity(
+                StackExchangeRedisCallsInstrumentation.ActivityName,
```

Comment:
Can we just pass the `name` directly here and then dont change displayname later in line 45?

Reply:
what we could do is in the first if, if name is null, set as the activityName. And, we would change to name in the StartActivity + remove the setter below.

Reply:
i will push soon.

Reply:
Just FYI I did it this way because .NET Framework HttpWebRequest & .NET Framework SqlClient ActivitySources also do it that way.

Reply:
should i change or should i maintain as it is right now?

Reply:
I always tend to vote for consistency. So leave as is, or change all three 🤣 Defer to @cijothomas.

Reply:
Since the name is readily available here, lets just use it.
The spec allows changing name after - and that is to be used in scenarios where name is not available when span begins. But in this case, we know the name, and lets just use it and dont change afterwards.

Reply:
ok. with that in mind, i will look for the other places and update as well.

Reply:
> Just FYI I did it this way because .NET Framework HttpWebRequest & .NET Framework SqlClient ActivitySources also do it that way.

We can revisit them if needed. I remember for sql we did that as we could save the cost associated with getting name from payload, only if activity is sampled in, right?